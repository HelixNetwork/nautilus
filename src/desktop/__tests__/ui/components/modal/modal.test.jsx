import React from 'react';
import { shallow } from 'enzyme';

import Modal from 'ui/components/modal/Modal';

const props = {
    children: 'Lorem ipsum',
    onClose: jest.fn(),
    inline: true,
    isOpen: false,
};

describe('Modal component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Modal {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Display content', () => {
        const wrapper = shallow(<Modal {...props} />);

        expect(wrapper.find('.content').text()).toEqual('Lorem ipsum');
    });

    // test('Modal hidden', () => {
    //     const wrapper = shallow(<Modal {...props} />);
    //     
    //     expect(wrapper.hasClass('hidden')).toBeTruthy();
    // });

    test('Modal visible', () => {
        const mockProps = Object.assign({}, props, { isOpen: true });
        const wrapper = shallow(<Modal {...mockProps} />);

        expect(wrapper.hasClass('hidden')).toBeFalsy();
    });

    // test('Modal variant is global', () => {
    //     const mockProps = Object.assign({}, props, { className: 'global' });
    //     const wrapper = shallow(<Modal {...mockProps} />);

    //     expect(wrapper.hasClass('global')).toBeTruthy();
    // });

    // test('Modal variant is confirm', () => {
    //     const mockProps = Object.assign({}, props, { className: 'confirm' });
    //     const wrapper = shallow(<Modal {...mockProps} />);

    //     expect(wrapper.hasClass('confirm')).toBeTruthy();
    // });
});
