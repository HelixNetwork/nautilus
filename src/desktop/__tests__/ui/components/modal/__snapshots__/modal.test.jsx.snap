// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Modal component Render the Model component 1`] = `
<div
  className="backdrop confirm hidden"
>
  <div
    className="wrapper"
  >
    <div
      className="content"
    >
      Lorem ipsum
    </div>
  </div>
</div>
`;
